unittests/test_meta.py

```diff
@@ -254,3 +256,91 @@ class AllowFinalOverride(Base):
 
         def foo(self):
             '''Overriding foo is now allowed.'''
+
+
+def test_callable_attributes(MyMeta):
+    '''Test issue #2113.
+
+    Setting a callable without the __name__ attribute would crash the
+    metaclass.
+    '''
+
+    class Callable:
+        def __call__(self):
+            pass
+
+    class Base(MyMeta):
+        f = Callable()
+
+
+def test_performance_function(MyMeta):
+    assert hasattr(MyMeta, '_rfm_perf_fns')
+
+    class Base(MyMeta):
+        @performance_function('units')
+        def perf_a(self):
+            pass
+
+        @performance_function('units')
+        def perf_b(self):
+            pass
+
+        def assert_perf_fn_return(self):
+            assert isinstance(
+                self.perf_a(), deferrable._DeferredPerformanceExpression
+            )
+
+    # Test the return type of the performance functions
+    Base().assert_perf_fn_return()
+
+    # Test the performance function set has been built correctly
+    assert len(Base._rfm_perf_fns) == 2
+    perf_fns = {'perf_a', 'perf_b'}
+    for fn in Base._rfm_perf_fns:
+        assert fn._rfm_perf_key in perf_fns
+        perf_fns.remove(fn._rfm_perf_key)
+
+    assert not perf_fns
```

Comment:
I think the following covers everything:

```suggestion
    assert {fn._rfm_perf_key for fn in Base._rfm_perf_fns} == {'perf_a', 'perf_b'}
```